recipes/magnum-integration/all/conanfile.py

```diff
@@ -0,0 +1,166 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration, ConanException
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class MagnumIntegrationConan(ConanFile):
+    name = "magnum-integration"
+    description = "Integration libraries for the Magnum C++11/C++14 graphics engine"
+    license = "MIT"
+    topics = ("magnum", "graphics", "rendering", "3d", "2d", "opengl")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://magnum.graphics"
+
+    settings = "os", "compiler", "build_type", "arch"
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_bullet": [True, False],
+        "with_dart": [True, False],
+        "with_eigen": [True, False],
+        "with_glm": [True, False],
+        "with_imgui": [True, False],
+        "with_ovr": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_bullet": True,
+        "with_dart": False,
+        "with_eigen": True,
+        "with_glm": True,
+        "with_imgui": True,
+        "with_ovr": False,
+    }
+    generators = "cmake", "cmake_find_package"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+
+    short_paths = True
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+    
+    def requirements(self):
+        self.requires("magnum/{}".format(self.version))
+        if self.options.with_bullet:
+            self.requires("bullet3/3.17")
+        if self.options.with_dart:
+            # FIXME: Add 'dart' requirement
+            raise ConanInvalidConfiguration("DART library is not available in ConanCenter (yet)")
+        if self.options.with_eigen:
+            self.requires("eigen/3.4.0")
+        if self.options.with_glm:
+            self.requires("glm/0.9.9.8")
+        if self.options.with_imgui:
+            self.requires("imgui/1.84.2")
+        if self.options.with_ovr:
+            # FIXME: Add 'ovr' requirement
+            raise ConanInvalidConfiguration("OVR library is not available in ConanCenter (yet)")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_STATIC"] = not self.options.shared
+        self._cmake.definitions["BUILD_STATIC_PIC"] = self.options.get_safe("fPIC", False)
```

Comment:
Default should be true since we delete it when shared (which makes it required)

```suggestion
        self._cmake.definitions["BUILD_STATIC_PIC"] = self.options.get_safe("fPIC", True)
```

Reply:
No actually you might be safe!
I missed "static" reviewing on mobile... shared here does not matter

Reply:
Given the sources of the library, it is already managed inside. It really doesn't matter the input value when `shared=True`:

```
if(NOT BUILD_STATIC OR BUILD_STATIC_PIC)
    set_target_properties(MagnumShadersObjects PROPERTIES POSITION_INDEPENDENT_CODE ON)
endif()
```